src/listener.js

```diff
@@ -58,18 +56,30 @@ module.exports = ({ handler, upgrader }, WebRTCStar, options = {}) => {
 
       channel.signal(offer.signal)
 
-      let conn
-      try {
-        conn = await upgrader.upgradeInbound(maConn)
-      } catch (err) {
-        log.error('inbound connection failed to upgrade', err)
-        return maConn.close()
-      }
-      log('inbound connection %s upgraded', maConn.remoteAddr)
+      channel.once('connect', async () => {
+        const maConn = toConnection(channel)
+        log('new inbound connection %s', maConn.remoteAddr)
+
+        let conn
+        try {
+          conn = await upgrader.upgradeInbound(maConn)
+        } catch (err) {
+          log.error('inbound connection failed to upgrade', err)
+          return maConn.close()
+        }
+
+        if (!conn.remoteAddr) {
+          try {
+            conn.remoteAddr = ma.decapsulateCode(CODE_P2P).encapsulate(`/p2p/${conn.remotePeer.toString()}`)
```

Comment:
Should this be the old style for now?

```suggestion
            conn.remoteAddr = ma.decapsulateCode(CODE_P2P).encapsulate(`/p2p/${conn.remotePeer.toB58String()}`)
```

Reply:
Yes, until we do the full migration to base32. I've committed the suggestion.